dao/src/main/java/org/wso2/testgrid/dao/repository/DeploymentPatternRepository.java

```diff
@@ -130,7 +131,9 @@ public DeploymentPattern findByPrimaryKey(String id) throws TestGridDAOException
                 "dp.PRODUCT_id = '" + productId + "' GROUP BY dp.id;";
         try {
             Query query = entityManager.createNativeQuery(queryStr, DeploymentPattern.class);
-            return query.getResultList();
+            @SuppressWarnings("unchecked")
```

Comment:
Any reason to suppress unchecked cast warning?

Reply:
Method returns List<DeploymentPattern>  and query.getResultList() return List. Therefore casting is required and can't eliminate the warning. Since this is type safe, added @SuppressWarnings("unchecked") annotation.